server/src/main/java/org/eclipse/openvsx/search/SearchConfig.java

```diff
@@ -34,17 +48,56 @@
     @Value("${ovsx.elasticsearch.password:}")
     String password;
 
+    @Value("${ovsx.elasticsearch.truststore:}")
+    String trustStore;
+
+    @Value("${ovsx.elasticsearch.truststorePassword:}")
+    String trustStorePassword;
+
     @Override
     public RestHighLevelClient elasticsearchClient() {
         var builder = ClientConfiguration.builder();
         var connected = Strings.isNullOrEmpty(searchHost)
                 ? builder.connectedToLocalhost()
                 : builder.connectedTo(searchHost);
-        var secure = useSsl ? connected.usingSsl() : connected;
+        var secure = useSsl ? connected.usingSsl(sslContext()) : connected;
         var authenticated = Strings.isNullOrEmpty(username) || Strings.isNullOrEmpty(password)
                 ? secure
                 : secure.withBasicAuth(username, password);
         return RestClients.create(authenticated.build()).rest();
     }
 
+    /**
+     * Returns a new trust store if {@link #trustStore} (and {@link #trustStorePassword}) properties 
+     * are non empty. Returns {@link SSLContext#getDefault() default} SSLContext otherwise.
+     */
+    private SSLContext sslContext() {
+        if (!Strings.isNullOrEmpty(trustStore)) {
+            var sslContextBuilder = SSLContextBuilder.create();
+            if (!Strings.isNullOrEmpty(trustStorePassword)) {
+                try {
+                    sslContextBuilder.loadTrustMaterial(new File(trustStore), trustStorePassword.toCharArray());
+                } catch(NoSuchAlgorithmException | KeyStoreException | CertificateException | IOException e) {
+                    logger.error("Unable to load password protected trust material " + trustStorePassword, e);
```

Comment:
Did you mean to print `trustStore` here? We should not log any password.

Reply:
Good catch. Fixed!